bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondBridgeHandler.java

```diff
@@ -0,0 +1,299 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.bondhome.internal.api.BPUPListener;
+import org.openhab.binding.bondhome.internal.api.BPUPUpdate;
+import org.openhab.binding.bondhome.internal.api.BondDeviceState;
+import org.openhab.binding.bondhome.internal.api.BondHttpApi;
+import org.openhab.binding.bondhome.internal.api.BondSysVersion;
+import org.openhab.binding.bondhome.internal.config.BondBridgeConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BondBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Sara Geleskie Damiano - Initial contribution
+ */
+@NonNullByDefault
+public class BondBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(BondBridgeHandler.class);
+
+    // Get a dedicated threadpool for the long-running listener thread.
+    // Intent is to not permanently tie up the common scheduler pool.
+    private final ScheduledExecutorService bondScheduler = ThreadPoolManager.getScheduledPool("bondBridgeHandler");
+    private @Nullable ScheduledFuture<?> listenerJob;
+    private final BPUPListener udpListener;
+    private final BondHttpApi api;
+
+    private @NonNullByDefault({}) BondBridgeConfiguration config;
+
+    private final Set<BondDeviceHandler> handlers = Collections.synchronizedSet(new HashSet<>());
+
+    public BondBridgeHandler(Bridge bridge) {
+        super(bridge);
+        udpListener = new BPUPListener(this);
+        api = new BondHttpApi(this);
+        logger.debug("Created a BondBridgeHandler for thing '{}'", getThing().getUID());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Not needed, all commands are handled in the {@link BondDeviceHandler}
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing the Bond bridge!");
+        config = getConfigAs(BondBridgeConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
+            initializeThing();
+        });
+    }
+
+    private void initializeThing() {
+        config = getConfigAs(BondBridgeConfiguration.class);
+        if (config != null) {
```

Comment:
If you work with a local variable, you get rid of the null check and the compiler warning.
```suggestion
        BondBridgeConfiguration localConfig = config = getConfigAs(BondBridgeConfiguration.class);
```

Reply:
removed